src/crystal/system/dir.cr

```diff
@@ -4,17 +4,18 @@ module Crystal::System::Dir
   #
   # Information about a directory entry.
   #
-  # In particular we only care about the name and whether its
-  # a directory or not to improve the performance of Dir.glob
-  # by avoid having to call File.info on every directory entry.
+  # In particular we only care about the name, whether it's a directory, and
+  # whether any hidden file attributes are set to improve the performance of
+  # `Dir.glob` by not having to call `File.info` on every directory entry.
   # If dir is nil, the type is unknown.
   # In the future we might change Dir's API to expose these entries
   # with more info but right now it's not necessary.
   struct Entry
     getter name
     getter? dir
+    getter? hidden
 
-    def initialize(@name : String, @dir : Bool?)
+    def initialize(@name : String, @dir : Bool?, @hidden : Bool)
```

Comment:
btw, why `@dir` is nilable?

Reply:
I don't get the reference.

Reply:
Line 10:
> # If dir is nil, the type is unknown.

Reply:
On POSIX a symlink is neither a (regular) file nor directory, you have to dereference it and hope that the target exists.

Reply:
Thanks for clarification!